### PR TITLE
Add WAF in Detection Mode to Dev and Prod

### LINKED ISSUE #3263: Enable WAF in detection mode
### Why are we doing this?
A recent vulnerability scan from a potentially malicious outside actor prompted an exploration into additional WAF implementations. We have some configurable items for this via Akamai already, but would like to experiment with Azure's offerings to see what we can gain.

### Action requested
- Enable geographic restrictions in "Detection" mode
- Enable OWASP managed ruleset in "Detection" mode.

### PR DESCRIPTION
**Pull request by rin-skylight**
## Related Issue or Background Info

- Resolves #3263 

## Changes Proposed

As a result of the destabilizing security situation in Europe, and the increased popularity of the SimpleReport system, SR has become an increasingly high-profile target for malicious actors. The use of a Web Application Firewall at the Application Gateway level can shield against most automated attacks, keeping our app and end users' data safe while freeing our engineers to keep adding new features.

This changeset performs the following actions:
- Upgrades the App Gateway SKU to use WAF_v2, instead of Standard_v2.
- Creates a new WAF module in TF.
- Creates a new WAF policy:
  - Leverage the experimental OWASP 3.1 ruleset to guard against the most common known attack vectors
  - Restrict all traffic originating outside of the United States and Canada
- Associates the new WAF policy, in detection mode, with `dev` and `prod`.

## Additional Information

- An upgrade to the WAF_v2 SKU is required for this to work. According to the Azure documentation, pricing should remain the same: https://azure.microsoft.com/en-us/pricing/details/application-gateway/ 
- `terraform plan` shows that the existing load balancer will be updated in-place, rather than completely destroyed.
- This has been deployed to `dev` for testing and verification.
- Documentation forthcoming regarding how to access WAF metrics from within Azure. Expect to see that go on the Wiki.

### Infrastructure
- [ ] **Consult the results of the `terraform-plan` job inside the "Terraform Checks" workflow run for this PR. Confirm that there are no unexpected changes!**

### Security
- [x] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [x] Any dependencies introduced have been vetted and discussed

## Cloud
- [x] DevOps team has been notified if PR requires ops support
- [x] If there are changes that cannot be tested locally, this has been deployed to our Azure `test`, `dev`, or `pentest` environment for verification

### CODE REVIEW

**Review by nickclyde — APPROVED** [review]
Thanks for your work on this Rin! Checked the logs in dev, looks like there were 22 requests detected from outside the US and Canada, with several from Russia and China. Definitely curious to see what is going on in production.

**Review by emmastephenson — APPROVED** [review]
Overall LGTM! My only concern is with Akamai, I don't want there to be any unexpected interactions. Do you know if Akamai is enabled on `dev` already?

File: ops/services/web_application_firewall/main.tf [hunk 1]
```diff
@@ -0,0 +1,36 @@
+resource "azurerm_web_application_firewall_policy" "sr_waf_policy" {
+  name                = "${var.name}-wafpolicy"
+  resource_group_name = var.resource_group_name
+  location            = var.resource_group_location
+
+  custom_rules {
+    name      = "US_Canada_Only"
```
Comment (jdorothy) [inline ops/services/web_application_firewall/main.tf:7]: I know this is only in detection mode but we'd want to confirm that enforcing this restriction is ok. I would think we'd prefer to go through Akamai here.

**Comment by jdorothy** [comment]
> Overall LGTM! My only concern is with Akamai, I don't want there to be any unexpected interactions. Do you know if Akamai is enabled on `dev` already?

Yes, Akamai is enabled in all envs.

**Review by jdorothy — APPROVED** [review]

File: ops/services/web_application_firewall/main.tf (hunk 1 above)
Comment (rin-skylight) [reply ops/services/web_application_firewall/main.tf:7]: Do you have any info on whether that's turned on/integrated at the moment? We've brought that up in previous syncs, but I'm unsure if we ever closed the loop. At last check, I thought that was something we needed to engage the CDC on, but I may be mistaken!

**Pull request merged**